app/src/main/java/com/sdpteam/connectout/authentication/GoogleLoginActivity.java

```diff
@@ -1,56 +1,73 @@
 import androidx.appcompat.app.AppCompatActivity;
 
-import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract;
-import com.sdpteam.connectout.R;
-import com.sdpteam.connectout.registration.CompleteRegistrationActivity;
-
 /**
  * Redirect automatically to next view (CompleteRegistrationActivity) if the user is already authenticated
  */
 public class GoogleLoginActivity extends AppCompatActivity {
-    private Authentication authentication = new GoogleAuth();
+    public static Authentication authentication = new GoogleAuth();
+    private Button loginBtn;
+    private ProgressBar progressBar;
+    private TextView infoText;
     private final ActivityResultLauncher<Intent> fireBaseSignInLauncher = registerForActivityResult(
             new FirebaseAuthUIActivityResultContract(),
             firebaseActivityResult -> {
                 redirectIfAuthenticated();
             });
 
-    void setAuthenticationService(Authentication a) {
-        // useful for mocking in tests
-        this.authentication = a;
-    }
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_google_login);
 
-        Button b = findViewById(R.id.loginButton);
-        b.setOnClickListener(v -> {
+        loginBtn = findViewById(R.id.loginButton);
+        loginBtn.setOnClickListener(v -> {
             fireBaseSignInLauncher.launch(authentication.buildIntent());
         });
 
+        progressBar = findViewById(R.id.progressBar);
+        progressBar.setVisibility(View.GONE);
+
+        infoText = findViewById(R.id.infoText);
+        infoText.setVisibility(View.GONE);
+
         redirectIfAuthenticated();
     }
 
     void redirectIfAuthenticated() {
         AuthenticatedUser authenticatedUser = authentication.loggedUser();
         if (authenticatedUser != null) {
+            loginBtn.setVisibility(View.GONE);
+
+            infoText.setVisibility(View.VISIBLE);
+            infoText.setText("Hello " + authenticatedUser.name + "\nyou will be redirected in few seconds");
+
+            progressBar.setVisibility(View.VISIBLE);
+
             navigateToSecondActivity(authenticatedUser);
         }
     }
 
     private void navigateToSecondActivity(AuthenticatedUser currentUser) {
-        Intent intent = new Intent(GoogleLoginActivity.this, CompleteRegistrationActivity.class);
-        String msg = currentUser.name + " \n" + currentUser.email;
-        intent.putExtra("loginInfo", msg);
-        startActivity(intent);
+        new ProfileFirebaseDataSource().fetchProfile(currentUser.uid).thenAccept(profile -> {
+            finish();
+            Intent intent = new Intent(GoogleLoginActivity.this, profile == null ? CompleteRegistrationActivity.class : DrawerActivity.class);
+            intent.putExtra("loginInfo", currentUser.name + " \n" + currentUser.email);
+            startActivity(intent);
+        });
```

Comment:
I would suggest against this approach as it breaks encapsulation, I would suggest adopting the MVVM paradigm.

Reply:
Well, here it is not a big deal, because this activity is very very small and does not have a viewmodel, it is a bit overkill